src/main/kotlin/org/rust/lang/core/completion/CompletionEngine.kt

```diff
@@ -84,7 +84,7 @@ fun RsCompositeElement.createLookupElement(scopeName: String): LookupElement {
             }
 
         is RsPatBinding -> base
-            .withTypeText(inferDeclarationType(this).let {
+            .withTypeText(this.type.let {
```

Comment:
Probably don't need explicit `this` anymore. Also usually any code with `let` can be made a tiny bit better by avoiding `let` :)